src/utils/documentPicker.ts

```diff
@@ -0,0 +1,318 @@
+import * as vscode from "vscode";
+import { AtelierAPI } from "../api";
+import { outputChannel } from ".";
+
+interface DocumentPickerItem extends vscode.QuickPickItem {
+  /** The full name of this item, including its parent(s). */
+  fullName: string;
+}
+
+function sodItemToDocumentPickerItem(
+  item: { Name: string; Type: number },
+  parent?: string,
+  parentPad?: number
+): DocumentPickerItem {
+  const result: DocumentPickerItem = { label: item.Name, fullName: item.Name };
+  // Add the icon
+  if (item.Type == 0) {
+    if (item.Name.endsWith(".inc")) {
+      result.label = "$(file-symlink-file) " + result.label;
+    } else if (item.Name.endsWith(".int") || item.Name.endsWith(".mac")) {
+      result.label = "$(note) " + result.label;
+    } else {
+      result.label = "$(symbol-misc) " + result.label;
+    }
+  } else if (item.Type == 10) {
+    result.label = "$(folder) " + result.label;
+  } else if (item.Type == 9) {
+    result.label = "$(package) " + result.label;
+  } else if (item.Type == 4) {
+    result.label = "$(symbol-class) " + result.label;
+  } else {
+    result.label = "$(symbol-file) " + result.label;
+  }
+  if (parent) {
+    // Update the full name and label padding if this is a nested item
+    let delim = ".";
+    if (parent.includes("/")) {
+      delim = "/";
+    }
+    result.fullName = parent + delim + item.Name;
+    result.label = " ".repeat(parentPad + 2) + result.label;
+    result.description = result.fullName;
+  }
+  if (item.Type == 9 || item.Type == 10) {
+    // Add the expand button if this is a package or directory
+    result.buttons = [
+      {
+        iconPath: new vscode.ThemeIcon("chevron-left"),
```

Comment:
I used `chevron-left` because it appears on the right side of item, where `chevron-right` is used on the left side of the items in VS Code's tree views. I figured that mirroring the entire thing was clearer than using the same icon on the other side of the items. This is the same as the project UI that I added last year, and it looks we had a similar conversation then: https://github.com/intersystems-community/vscode-objectscript/pull/927#issuecomment-1098269284.

Reply:
Seems I still find it confusing then. But I'm not going to dig my heels in.